### PR TITLE
refactor: share modal ui to open separate form for squad

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Remove the Markdown input from the ShareModal.
- When clicking a Squad icon, we will then open the modal to share a post to Squad.
- Since the ShareModal was not used in the LazyModal enum, we won't have to make any changes with the LazyModal itself since only one instance is opened.

Preview:

https://github.com/dailydotdev/apps/assets/13744167/0b2bc640-3775-489b-8203-01e12181df82

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1794 #done

### CODE REVIEW

**Review by rebelchris — COMMENTED** [review]
The only thing I might be overlooking how does the first modal get closed when the second opens?

**Comment by sshanzel** [comment]
> The only thing I might be overlooking how does the first modal get closed when the second opens?

Do we have to close the first one when we just opened the second one? My concern is if the user cancels sharing with the Squad.

**Comment by rebelchris** [comment]
There is no way to go back anyway right?
I think the cleanest is to simply close it while opening the new one.

What would be a flow of having it still? I can try it locally to see what happens.

Error will look into on Monday

**Comment by sshanzel** [comment]
> There is no way to go back anyway right?

Closing the modal for input, then you should be back to the original modal.

> What would be a flow of having it still? I can try it locally to see what happens.

Like you wanted it to be shared on some Squad but doubted and didn't proceed. Instead, I want to copy the link to be shared through some other platforms.

**Comment by rebelchris** [comment]
> > There is no way to go back anyway right?
> 
> Closing the modal for input, then you should be back to the original modal.
> 
> > What would be a flow of having it still? I can try it locally to see what happens.
> 
> Like you wanted it to be shared on some Squad but doubted and didn't proceed. Instead, I want to copy the link to be shared through some other platforms.

Alright, checked and let's leave like this it works

**Comment by sshanzel** [comment]
The tests have passed now 🚀

File: packages/shared/src/components/modals/ShareModal.spec.tsx [hunk 1]
```diff
@@ -40,43 +41,33 @@ const renderComponent = (
...
+        isOpen
+        onRequestClose={onRequestClose}
+        ariaHideApp={false}
+      />
+    </TestBootProvider>,
   );
 };
 
 describe('ShareModal Test Suite:', () => {
-  it('should render the article preview', async () => {
```
Comment (sshanzel) [inline packages/shared/src/components/modals/ShareModal.spec.tsx:71]: Not used here anymore. Different modal, different tests.

**Review by rebelchris — APPROVED** [review]
Let's go

**Pull request merged**